### ISSUE TITLE
Aria-describedby for listbox hidden inputs

### ISSUE THREAD

**Issue by Nargonath** [tailwindlabs/headlessui#2994]
**What package within Headless UI are you using?**

@headlessui/react

**What version of that package are you using?**

v1.7.18

**What browser are you using?**

Brave (chromium based)

**Reproduction URL**

https://github.com/Nargonath/listbox-describedby-bug

**Describe your issue**

The `aria-describedby` attribute assigned to the `Listbox` component is assigned to the `div` parent element but I believe it should be assigned to the `input[type="hidden"]` when a `name` is passed to `Listbox` for accessibility purposes. Let me know if I'm wrong though.

**Comment by RobinMalfait** [comment]
Hey!

The hidden input fields are an implementation detail to make the component work in native `<form>` elements. You are also not directly interacting with the hidden input fields so even screenreaders won't benefit from pointing the `aria-describedby` to the hidden input fields.

However, you are interacting with the `Listbox.Button` and the `Listbox.Options`. These are the components that screenreaders and other assistive technology would interact with and announce to the user.

In your case, I think the best thing to do would be to move the `aria-described` by to the actual button. That way you get this information before opening the `Listbox`. 

I would not add it to the `Listbox.Options` or `Listbox.Option` because then the information would be announced every time you move to a different item.

Concretely, I think you can apply this diff:

```diff
  export function ListBox() {
    const [selectedPerson, setSelectedPerson] = React.useState(people[0]);

    return (
      <>
        <Listbox
          as="div"
          value={selectedPerson}
          onChange={setSelectedPerson}
          name="test-list"
-         aria-describedby="error-message"
        >
-         <Listbox.Button>
+         <Listbox.Button aria-describedby="error-message">
            {selectedPerson.name}
          </Listbox.Button>
          <Listbox.Options>
            {people.map((person) => (
              <Listbox.Option
                key={person.id}
                value={person}
                disabled={person.unavailable}
              >
                {person.name}
              </Listbox.Option>
            ))}
          </Listbox.Options>
        </Listbox>

        <p id="error-message">some info</p>
      </>
    );
  }
``` 

Hope this helps!

**Comment by Nargonath** [comment]
Ok Thank you Robin.
